### PR TITLE
test: add e2e tests for Flatcar Linux

### PR DESCRIPTION
**Pull request by ccojocar**
Signed-off-by: Cosmin Cojocar <gcojocar@adobe.com>

#### Does this PR introduce a user-facing change?

```release-note
Add E2E tests for cluster running on Flatcar Linux
```

### CODE REVIEW

**Comment by ccojocar** [comment]
@saschagrunert @janetkuo @JAORMX The base e2e tests passed, please could you have a look over my changes?

I am trying to enable the eBPF profile recording in a different pull request. Thanks

File: hack/ci/Vagrantfile-flatcar [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    sh.inline = <<~SHELL
+      set -euxo pipefail
+
+      # Install Go
+      GO_VERSION=1.18
+      mkdir /opt/go
+      curl -sSfL -o- https://golang.org/dl/go$GO_VERSION.linux-amd64.tar.gz | tar xfz - -C /opt
+
+      # Setup Networking 
+      systemctl enable docker
```
Comment (JAORMX) [inline hack/ci/Vagrantfile-flatcar:42]: Why do we need to enable Docker? Isn't containerd enough?

Comment (saschagrunert) [inline hack/ci/Vagrantfile-flatcar:42]: We can't update those tests to Kubernetes v1.24.0 when they removed dockershim, can we use CRI-O or containerd here?

Comment (ccojocar) [reply hack/ci/Vagrantfile-flatcar:42]: CRI-O is not yet available in Flatcar. I tried to install it manually and the binary was crashing. I am not sure what's the reason for that. I can update it when they enable it.

File: hack/ci/Vagrantfile-flatcar [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+      # Disable kernel print rate limiting for syslog messaging
+      sysctl -w kernel.printk_ratelimit=0
+      sysctl -w kernel.printk_ratelimit_burst=0
+
+      # Import the test image for security profiles operator 
+      docker load -i /vagrant/image.tar
+
+      # Install cri-o
+      # /vagrant/hack/ci/install-cri-o.sh
```
Comment (JAORMX) [inline hack/ci/Vagrantfile-flatcar:87]: If this is not used, could you remove this? It's confusing for readers

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: hack/ci/Vagrantfile-flatcar (hunk 1 above)
Comment (JAORMX) [reply hack/ci/Vagrantfile-flatcar:42]: `containerd` is available by default in Flatcar though.

File: hack/ci/Vagrantfile-flatcar [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+      chown -R core:core $HOME/.kube
+      export KUBECONFIG=/etc/kubernetes/admin.conf
+
+      # Configure cluster
+      kubectl taint nodes --all node-role.kubernetes.io/master-
+
+      # Install CNI
+      kubectl create -f https://docs.projectcalico.org/manifests/tigera-operator.yaml
+      kubectl apply -f /vagrant/hack/ci/flatcar-cni-plugin.yaml
```
Comment (JAORMX) [inline hack/ci/Vagrantfile-flatcar:115]: Is there a reason why Calico is needed? Could we simply use the `bridge` SNI plugin like the other tests do?

Comment (ccojocar) [reply hack/ci/Vagrantfile-flatcar:115]: Somehow the cluster was not becoming ready without Calico. This is why I enabled it.

File: hack/ci/Vagrantfile-flatcar (hunk 1 above)
Comment (ccojocar) [reply hack/ci/Vagrantfile-flatcar:42]: I'll switch if off and try again only with containerd. Let's see if it works.

File: hack/ci/Vagrantfile-flatcar (hunk 2 above)
Comment (ccojocar) [reply hack/ci/Vagrantfile-flatcar:87]: I removed these commented lines.

File: hack/ci/Vagrantfile-flatcar (hunk 1 above)
Comment (ccojocar) [reply hack/ci/Vagrantfile-flatcar:42]: I'll try again only with containerd.

Comment (ccojocar) [reply hack/ci/Vagrantfile-flatcar]: The e2e tests use now only containerd.

**Comment by ccojocar** [comment]
@saschagrunert @JAORMX The e2e tests passed now with containerd.

**Review by JAORMX — APPROVED** [review]
/approve

Thanks for working on this! I really dig that we now have Flatcar AND containerd coverage

**Comment by saschagrunert** [comment]
Great work @ccojocar :pray:!

**Pull request merged**